mitodl/mit-open
# Reinstate head metadata

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/5410

### Description (What does it do?)
This PR re-introduces meta tags (title and social share tags where applicable. For public/unauthenticated views the meta tags are rendered server-side. After some discussion with @jonkafton it seems SSR for meta tags may not be possible for authenticated views (in this case we default to regular js generated tags).

### How can this be tested?
Checkout this branch and browse around. On search pages, resource drawer pages, the home page (all public views) you should be able to view the source and see that meta tags and the title tags are embedded in the server-sent markup. For other pages requiring authentication such as the settings dashboard etc you should still see a valid title however it is written after the page renders. 

### Additional Context
This PR also includes fixes for 2 issues I encountered that were simple to patch - [an error I saw on the search page](https://github.com/mitodl/mit-open/pull/1572/files#diff-8ad4a04533240b5a82bca338b1fb32c3c878965c03bd63f7d00f5d4cbbd02e8dR67) due to (afaik) how nextjs expects things to be wrapped and a [fix for broken navigation](https://github.com/mitodl/mit-open/pull/1572/files#diff-91beb948a052e5c93f515854781d5ca7e14bc9b32f5a761fcdbf6d2e8fc38a87R316
) on the settings page.

This PR also does not include tests - this will be separately outside of this PR since we need to figure out how to properly test server-side components.
